lib/logstash/outputs/elastic_app_search.rb

```diff
@@ -11,10 +11,15 @@ class LogStash::Outputs::ElasticAppSearch < LogStash::Outputs::Base
   config :api_key, :validate => :password, :required => true
   config :timestamp_destination, :validate => :string
   config :document_id, :validate => :string
+  config :service_type, :validate => :string
```

Comment:
Looking at the [java client api docs](https://github.com/swiftype/swiftype-app-search-java#using-with-app-search-managed-deploys) it seems that the parameters for the client are:

#### for hosted app search:
* hostIdentifier = "host-c5s2mj"
* apiKey = "private-mu75psc5egt9ppzuycnc2mc3"

#### for self managed app search:
* apiKey = "private-mu75psc5egt9ppzuycnc2mc3"
* baseUrl = "http://localhost:3002/api/as/v1/"

So I'm thinking that instead of having this option that mimics a boolean flag, we could add the url option and remove the required flag from the `:host`, as such:
```
config :host, :validate => :string
config :url, :validate => :url
```

Then in the constructor we validate which one is configured and instantiate the Client accordingly, or error out if neither or both were set. What do you think? if you agree I can perform these changes and carry the PR forward.

Reply:
Sounds good to me! Thanks!